sunpy/coordinates/frames.py

```diff
@@ -0,0 +1,374 @@
+"""
+Common solar physics coordinate systems.
+
+This submodule implements various solar physics coordinate frames for use with
+the `astropy.coordinates` module.
+"""
+
+# NumPy import
+import numpy as np
+
+# Astropy imports
+from astropy import units as u
+from astropy.coordinates.representation import (CartesianRepresentation,
+                                                UnitSphericalRepresentation,
+                                                SphericalRepresentation)
+from astropy.coordinates.baseframe import (BaseCoordinateFrame,
+                                           RepresentationMapping)
+from astropy.coordinates import FrameAttribute
+
+# SunPy imports
+from sunpy import sun  # For Carrington rotation number
+from representation import (SphericalWrap180Representation,
+                            UnitSphericalWrap180Representation)
+
+from .frameattributes import TimeFrameAttributeSunPy
+
+RSUN_METERS = sun.constants.constant('radius').si.to(u.m)
+DSUN_METERS = sun.constants.constant('mean distance').si.to(u.m)
+
+__all__ = ['HeliographicStonyhurst', 'HeliographicCarrington',
+           'Heliocentric', 'Helioprojective', 'HelioprojectiveRadial']
+
+
+class HeliographicStonyhurst(BaseCoordinateFrame):
+    """
+    A coordinate or frame in the Stonyhurst Heliographic
+    system.
+
+    This frame has its origin at the solar centre and the north pole above the
+    solar north pole, and the zero line on longitude pointing towards the
+    Earth.
+
+    Parameters
+    ----------
+    representation: `~astropy.coordinates.BaseRepresentation` or None
+        A representation object or None to have no data.
+    lon: `Angle` object.
+        The longitude for this object (``lat`` must also be given and
+        ``representation`` must be None).
+    lat: `Angle` object.
+        The latitude for this object (``lon`` must also be given and
+        ``representation`` must be None).
+    rad: `astropy.units.Quantity` object.
+        This quantity holds the radial distance. If not specified, it is, by
+        default, the solar radius. Optional, must be keyword
```

Comment:
the solar radius, as defined by who? Maybe explicitly state the value and reference for that value?